src/Catalyst.Core.Modules.Consensus/Deltas/Building/TransactionRetrieverStep.cs

```diff
@@ -0,0 +1,131 @@
+
+        public TransactionRetrieverStep(IDeltaTransactionRetriever retriever, ILogger logger)
+        {
+            _retriever = retriever ?? throw new ArgumentNullException(nameof(retriever));
+            _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+        }
+
+        public void Execute(DeltaBuilderContext context)
+        {
+            IList<PublicEntry> allTransactions = _retriever.GetMempoolTransactionsByPriority();
+
+            Guard.Argument(allTransactions, nameof(allTransactions))
+               .NotNull("Mempool content returned null, check the mempool is actively running");
+
+            IList<PublicEntry> includedTransactions = GetValidTransactionsForDelta(allTransactions);
+            context.Transactions = includedTransactions;
+        }
+
+        private static bool IsTransactionOfAcceptedType(PublicEntry transaction) { return transaction.IsPublicTransaction || transaction.IsContractCall || transaction.IsContractDeployment; }
+
+        /// <summary>
+        ///     Gets the valid transactions for delta.
+        ///     This method can be used to extract the collection of transactions that meet the criteria for validating delta.
+        /// </summary>
+        private IList<PublicEntry> GetValidTransactionsForDelta(IList<PublicEntry> allTransactions)
+        {
+            //lock time equals 0 or less than ledger cycle time
+            //we assume all transactions are of type non-confidential for now
+
+            List<PublicEntry> validTransactionsForDelta = new List<PublicEntry>();
+            List<PublicEntry> rejectedTransactions = new List<PublicEntry>();
+
+            int allTransactionsCount = allTransactions.Count;
+            for (int i = 0; i < allTransactionsCount; i++)
+            {
+                PublicEntry currentItem = allTransactions[i];
+                if (!IsTransactionOfAcceptedType(currentItem))
+                {
+                    rejectedTransactions.Add(currentItem);
+                    continue;
+                }
+
+                validTransactionsForDelta.Add(currentItem);
+            }
+
+            validTransactionsForDelta.Sort(AveragePriceComparer.InstanceDesc);
+
+            ulong totalLimit = 0UL;
+            int allValidCount = validTransactionsForDelta.Count;
+            int rejectedCountBeforeLimitChecks = rejectedTransactions.Count;
+            for (int i = 0; i < allValidCount; i++)
+            {
+                PublicEntry currentItem = validTransactionsForDelta[i];
+                ulong remainingLimit = DeltaGasLimit - totalLimit;
+                if (remainingLimit < MinTransactionEntryGasLimit)
+                {
+                    for (int j = i; j < allValidCount; j++)
+                    {
+                        currentItem = validTransactionsForDelta[j];
+                        rejectedTransactions.Add(currentItem);
+                    }
+
+                    break;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Refactor the code in order to remove this break statement.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4979613)